### PR TITLE
Bump clap and other minor dependencies

### PR DESCRIPTION
**Pull request by SomeoneToIgnore**
Follow-up of https://github.com/neondatabase/neon/pull/2618, that bumps `clap` and the rest of the dependencies.

### CODE REVIEW

File: proxy/src/main.rs [hunk 1]
```diff
@@ -45,52 +45,122 @@ async fn main() -> anyhow::Result<()> {
...
             Arg::new("mgmt")
                 .short('m')
                 .long("mgmt")
-                .takes_value(true)
                 .help("listen for management callback connection on ip:port")
                 .default_value("127.0.0.1:7000"),
         )
         .arg(
             Arg::new("http")
-                .short('h')
```
Comment (SomeoneToIgnore) [inline proxy/src/main.rs:75]: cc @funbringer 

I've removed this parameter due to the new `cli` test started failing below.
It complained, that this `-h` conflicts with the short form of `--help`.

LMK if there's anything else you want to do about this.

**Review by koivunej — APPROVED** [review]
Cannot see anything obviously wrong, but the clap changes with the refactorings are a lot to look at. But as the test suite works, those must be good.

File: proxy/src/main.rs (hunk 1 above)
Comment (funbringer) [reply proxy/src/main.rs:75]: Sure! To be honest, I'd drop al shorthands.

**Review by funbringer — APPROVED** [review]

File: proxy/src/main.rs (hunk 1 above)
Comment (SomeoneToIgnore) [reply proxy/src/main.rs:75]: I've reached my breaking change limit recently and prefer to do less of those for a while, so will leave the others as is so far, thank you for confirming 🙂

**Pull request merged**